ci-operator/step-registry/ibmcloud/provision/dedicatedhost/dynamic/ibmcloud-provision-dedicatedhost-dynamic-commands.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+#Random select the dedicated host profile and based on the profile get the vm type.
+#Known Issues: 
+# OCPBUGS-5906 [IPI-IBMCloud] the worker nodes are “Provisioning” when using the dedicated host in the Default group
+# OCPBUGS-18925 [IPI-IBMCloud] ] fail to use vx2d-host-176x2464 profile with dedicated host in install-config
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    region="${LEASED_RESOURCE}"
+    export region
+    echo "Try to login..."
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+#random selete the dh profile, OCPBUGS-18925 blocked use "very-high-memory"
+#if need specify the dh profile to test, just change the return value.
+function randomGetDHProfile {
+    local all_profiles profiles selected_id proTotal
+    all_profiles=$(${IBMCLOUD_CLI} is dhps --output JSON | jq -r '.[] | select(.family != "very-high-memory") | .name')
+    mapfile -t profiles <<< "${all_profiles}"
+    proTotal=${#profiles[@]}
+    selected_id=$((RANDOM % ${proTotal}))
+    echo ${profiles[$selected_id]}
+}
+
+## get the vm instance profile with the dh profile header
+function getVMTypewithProfile() {
+    local profileHeader=$1
+    local all_profiles str1 str2
+    all_profiles=$(${IBMCLOUD_CLI} is instance-profiles -q | awk '(NR>1) {print $1}')
+    str1=${all_profiles##*$profileHeader}
+    str2=$(echo ${str1}|cut -d " " -f1)
+    echo ${profileHeader}${str2}
+}
+
+#####################################
+##############Initialize#############
+#####################################
+ibmcloud_login
+
+#####################################
+##random select the dedicated host profile ##
+#####################################
+
+echo "$(date -u --rfc-3339=seconds) - Random select the Dedicated Host..."
+
+dhProfileMaster=$(randomGetDHProfile)
+echo "random selected ${dhProfileMaster} for master nodes"
+class1=${dhProfileMaster%%-*}
+
+dhProfileWorker=$(randomGetDHProfile)
+echo "random selected ${dhProfileWorker} for worker nodes"
+class2=${dhProfileWorker%%-*}
+
+masterProfile=$(getVMTypewithProfile "${class1}-8x")
+workerProfile=$(getVMTypewithProfile "${class2}-4x")
+
+if [[ -z ${DEDICATEDHOST_ZONE} ]]; then
```

Comment:
`DEDICATEDHOST_ZONE` is already prompted as ENV var, so it would be never empty here.